src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationServiceTest.java

```diff
@@ -404,6 +409,23 @@ void shouldValidateJurisdictionCodesWithDuplicatesCodesAndExistenceJudgement() {
         assertEquals(JURISDICTION_CODES_DELETED_ERROR + "[CCP, ADG]", errors.get(1));
     }
 
+    @Test
+    void shouldValidateJurisdictionCodesForDisposedHearingWithSameDisposalDateOfJurisdiction() {
+        List<String> errors = new ArrayList<>();
+        eventValidationService.validateJurisdiction(caseDetails23.getCaseData(), errors);
+
+        assertEquals(0, errors.size());
+    }
+
+    @Test
+    void shouldHaveValidationErrorForJurisdictionCodesOfDisposedHearingWithDifferentDisposalDateOfJurisdiction() {
+        List<String> errors = new ArrayList<>();
+        eventValidationService.validateJurisdiction(caseDetails24.getCaseData(), errors);
+
+        assertEquals(1, errors.size());
+        assertEquals(String.format(DISPOSAL_DATE_HEARING_DATE_MATCH, "DDA"), errors.get(0));
+    }
```

Comment:
Thank you for adding the tests. I'd prefer to see you re-using existing case data and making the exact modifications within the test itself to make the method throw an error. Looking at these two tests alone its hard to see what input data changed that would lead to different output.

To see what exactly changed, I ran those two files through a diffchecker and there's only 1 field difference between them. Instead of adding a `caseDetailsTest24.json` file, It would be more appropriate here to re-use `caseDetailsTest23.json` and modify that field within the test function itself (like `caseData.getJurCodesCollection.get(0).getValue().setDisposalDate("2024-02-26")`).

But please compare this with one of the other test case details files, perhaps `caseDetailsTest23.json` is similar enough to `caseDetailsTest22.json` or another one?

Reply:
Thanks @JackReevies as we talked on the slack it is resolved by removing caseDetailsTest24.json. Unfortunately other existing json files doesn't match with the requirements that is why I was unable to use them. I agree with you that there should be only one json file and we should change the object values in test methods. But it would take a long time to correct it. That is why we need to handle it on a new task.